File: solutions/java/src/onlinestockbrokeragesystem/Order.java
package onlinestockbrokeragesystem;

public abstract class Order {
    protected final String orderId;
    protected final Account account;
    protected final Stock stock;
    protected final int quantity;
    protected final double price;
    protected OrderStatus status;

    public Order(String orderId, Account account, Stock stock, int quantity, double price) {
        this.orderId = orderId;
        this.account = account;
        this.stock = stock;
        this.quantity = quantity;
        this.price = price;
        this.status = OrderStatus.PENDING;
    }

    public abstract void execute();
}

File: solutions/csharp/onlinestockbrokeragesystem/Order.cs
namespace OnlineStockBrokerageSystem
{
    public abstract class Order
    {
        protected readonly string OrderId;
        protected readonly Account Account;
        protected readonly Stock Stock;
        protected readonly int Quantity;
        protected readonly double Price;
        protected OrderStatus Status;

        public Order(string orderId, Account account, Stock stock, int quantity, double price)
        {
            OrderId = orderId;
            Account = account;
            Stock = stock;
            Quantity = quantity;
            Price = price;
            Status = OrderStatus.Pending;
        }

        public abstract void Execute();
    }
}